torch/nn/parallel/distributed.py

```diff
@@ -1085,10 +1168,21 @@ def _run_ddp_forward(self, *inputs, **kwargs):
             with self._inside_ddp_forward():
                 return self.module(*inputs, **kwargs)
 
+    def _clear_grad_buffer(self):
+        if self._grad_buffer is not None:
+            for index, param in enumerate(self._delay_all_reduce_params):
+                param.grad = self._grad_views[index]
+            self._grad_buffer.zero_()
```

Comment:
just want to have a head up or you can add a comment here: "making param.grad points to the grad buffers before backward" is based on the assumption that the grad accumulation is done in place in autograd engine, for some edge cases, if the grad accumulation in autograd engine is not in place, then the param.grad and grad buffers are detached

Reply:
I added a comment here to clarify this implicit assumption.